packages/amplify-codegen-e2e-tests/src/__tests__/build-app-android.test.ts

```diff
@@ -49,11 +74,38 @@ describe('build app - Android', () => {
     }
   });
 
-  it('fails build with syntax error', async () => {
+  // not supported with conflict resolution enabled
+  [
+    ['v2-cyclic-has-one-dependency', schemas['v2-cyclic-has-one-dependency']],
+    ['v2-cyclic-has-many-dependency', schemas['v2-cyclic-has-many-dependency']],
+  ].forEach(([schemaName, schema]) => {
+    // @ts-ignore
+    it(`builds models with ${schemaName}: ${schema.description}`, async () => {
+      // @ts-ignore
+      const schemaText = `input AMPLIFY { globalAuthRule: AuthRule = { allow: public } }\n${schema.sdl}`;
+      updateApiSchemaWithText(projectRoot, apiName, schemaText);
+      await generateModels(projectRoot);
+      await androidBuild(projectRoot, { ...config });
+    });
+  });
+
+  it('fails build with syntax error in models', async () => {
     // @ts-ignore
     updateApiSchemaWithText(projectRoot, apiName, Object.values(schemas)[0].sdl);
+    await apiGqlCompile(projectRoot);
     await generateModels(projectRoot);
     await writeFileSync(path.join(projectRoot, modelDir, 'AmplifyModelProvider.java'), 'foo\nbar');
     await expect(androidBuild(projectRoot, { ...config })).rejects.toThrowError();
   });
+
+  it('fails build with syntax error in statements', async () => {
+    // @ts-ignore
+    updateApiSchemaWithText(projectRoot, apiName, Object.values(schemas)[0].sdl);
+    await apiGqlCompile(projectRoot);
+    await generateModels(projectRoot);
+    await generateStatementsAndTypes(projectRoot);
+    writeFileSync(path.join(projectRoot, statementsDir, 'mutations.graphql'), 'foo\nbar'),
+      expect(() => parse(readFileSync(path.join(projectRoot, statementsDir, 'mutations.graphql'), 'utf8'))).toThrowError();
+    await androidBuild(projectRoot, { ...config });
```

Comment:
this build is expected to fail right? Since we're adding invalid GQL.

Reply:
The android compilation won't fail in this case because the GraphQL statements are generated with raw GraphQL syntax. Line 108 covers the failure. Line 109 isn't really needed in this case.

Reply:
If it's not needed, is there a reason we don't just remove it?